### PR TITLE
Wait until pods are deleted in the initial cleanup

### PR DESCRIPTION
**Pull request by petr-muller**
Fixes: https://github.com/openshift/release/issues/1335 / DPTP-185

To clean up residual template instances and their pods from a possible
previous execution of ci-operator, `template.go:Run()` deletes the
template instance. The comment in the code indicates this does not block
on cleaning up template instance's `Pods`, so the cleanup code waits on
these being deleted after it deletes the template instance.

Previously, the cleanup waited on Pod completion, reusing the code being
used *after* execution to properly collect logs and wait until
everything shuts down gracefully. For some failure cases, this led to
confusing log output being printed during the cleanup, as well as ending
cleanup before the `Pods` were completely deleted. In such case, the
following template instance creation failed because pods to be created
still existed from previous execution. This change makes the cleanup
wait until the `Pods` are actually deleted, and also does not call any
of the code waiting on proper finish and colleting the log output
(because this output belongs to the previous exection anyway).

### CODE REVIEW

File: pkg/steps/template.go [hunk 1]
```diff
@@ -475,7 +476,7 @@ func waitForCompletedTemplateInstanceDeletion(templateClient templateclientset.T
 	for _, ref := range instance.Status.Objects {
 		switch {
 		case ref.Ref.Kind == "Pod" && ref.Ref.APIVersion == "v1":
-			waitForPodCompletion(podClient, ref.Ref.Name, nil)
+			waitForPodDeletion(podClient, ref.Ref.Name, "")
```
Comment (stevekuznetsov) [inline pkg/steps/template.go]: Should pass `ref.Ref.UID`?

Comment (petr-muller) [reply pkg/steps/template.go]: Right.

**Comment by petr-muller** [comment]
/hold

I'll add a condition to eventually finish the wait loop even if the deletion does not succeed for some reason and also propagate and fail on errors (or at the minimum, log them).

**Comment by petr-muller** [comment]
or not

/meow

**Pull request merged**